include/net/coap.h

```diff
@@ -233,7 +233,8 @@ typedef int (*coap_reply_t)(const struct coap_packet *response,
  */
 struct coap_pending {
 	struct sockaddr addr;
-	s32_t timeout;
+	u32_t t0;
+	u32_t timeout;
```

Comment:
We could convert coap to use internally the k_timeout_t, which would simplify the timeout handling a lot.

Reply:
ACK, will investigate.

Reply:
@jukkar Actually, I did some research, but I fail to see how switching to `k_timeout_t` would simplify implementation in this particular case. This could be my misunderstanding of `k_timeout_t` API, but for instance I miss a function to compare two timeouts to get the "earlier" one, or to initialize `k_timeout_t` structure with the current time plus some offset. Of course this could be done by reaching the internal fields of this struct, but I think that's not the point here (and this would break compability with legacy API which I assue will remain here for some time).

If you have in your mind how we could rework this please let me know, otherwise I think we should keep this as it is.

Reply:
Lets leave the current logic as it is now as it works ok. I was just looking the code and it looked complex and was hoping we could simplify things.